src/mapbuffer.cpp

```diff
@@ -130,20 +130,22 @@ void mapbuffer::save( bool delete_after_save )
         // A segment is a chunk of 32x32 submap quads.
         // We're breaking them into subdirectories so there aren't too many files per directory.
         // Might want to make a set for this one too so it's only checked once per save().
-        std::stringstream segment_path;
+        std::stringstream dirname;
         tripoint segment_addr = overmapbuffer::omt_to_seg_copy( om_addr );
-        segment_path << map_directory.str() << "/" << segment_addr.x << "." <<
+        dirname << map_directory.str() << "/" << segment_addr.x << "." <<
                      segment_addr.y << "." << segment_addr.z;
-        assure_dir_exist( segment_path.str().c_str() );
 
         std::stringstream quad_path;
-        quad_path << segment_path.str() << "/" << om_addr.x << "." <<
+        quad_path << dirname.str() << "/" << om_addr.x << "." <<
                   om_addr.y << "." << om_addr.z << ".map";
 
         // delete_on_save deletes everything, otherwise delete submaps
         // outside the current map.
-        save_quad( quad_path.str(), om_addr, submaps_to_delete,
-                   delete_after_save || om_addr.z != g->levz ||
+#ifndef ZLEVELS
+        delete_after_save |= om_addr.z != g->levz;
```

Comment:
I PR'd a fix here: #11441

The bug doesn't cause any save corruption or data loss, but it can cause the game to crash after the save finishes.